GNUmakefile

```diff
@@ -11,7 +25,16 @@ test: fmtcheck
 	echo $(TEST) | \
 		xargs -t -n4 go test $(TESTARGS) -timeout=30s -parallel=4
 
-testacc: fmtcheck
+testacc:
+	@:$(call check_defined, \
+						ARM_CLIENT_ID \
+						ARM_CLIENT_SECRET \
+						ARM_SUBSCRIPTION_ID \
+						ARM_TENANT_ID \
+						ARM_TEST_LOCATION \
+						ARM_TEST_LOCATION_ALT, \
+				"ARM_CLIENT_ID ARM_CLIENT_SECRET ARM_SUBSCRIPTION_ID ARM_TENANT_ID ARM_TEST_LOCATION and ARM_TEST_LOCATION_ALT must be set")
+	@$(MAKE) fmtcheck
```

Comment:
so there's already a check in the codebase for these variables, which should fail when these aren't specified (but only when running the Acceptance Tests, not any unit tests): https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/provider_test.go#L37

```
$ TF_ACC=1 go test ./azurerm -v -timeout 120m -run=TestAccAzureRMResourceGroup
=== RUN   TestAccAzureRMResourceGroup_importBasic
--- FAIL: TestAccAzureRMResourceGroup_importBasic (0.00s)
	provider_test.go:49: `ARM_CLIENT_ID` must be set for acceptance tests!
=== RUN   TestAccAzureRMResourceGroup_basic
--- FAIL: TestAccAzureRMResourceGroup_basic (0.00s)
	provider_test.go:49: `ARM_CLIENT_ID` must be set for acceptance tests!
=== RUN   TestAccAzureRMResourceGroup_disappears
--- FAIL: TestAccAzureRMResourceGroup_disappears (0.00s)
	provider_test.go:49: `ARM_CLIENT_ID` must be set for acceptance tests!
=== RUN   TestAccAzureRMResourceGroup_withTags
--- FAIL: TestAccAzureRMResourceGroup_withTags (0.00s)
	provider_test.go:49: `ARM_CLIENT_ID` must be set for acceptance tests!
FAIL
exit status 1
FAIL	github.com/terraform-providers/terraform-provider-azurerm/azurerm	0.027s
```